doc/source/ray-logging.rst

```diff
@@ -0,0 +1,132 @@
+Ray Logging
+===========
+This document will explain Ray's logging system and its best practices.
+
+Driver logs
+~~~~~~~~~~~
+An entry point of Ray applications that calls `ray.init(address='auto')` or `ray.init()` is called a driver.
+All the driver logs are handled in the same way as normal python programs. 
+
+Worker logs
+~~~~~~~~~~~
+Ray's tasks or actors are executed remotely within Ray's worker processes. Ray has special support to improve the visibility of logs produced by workers.
+- By default, all of the tasks/actors stdout and stderr are redirected to the worker log files. Check out :ref:`Logging directory structure <logging-directory-structure>` to learn how Ray's logging directory is structured.
+- By default, all of the tasks/actors stdout and stderr that is redirected to worker log files are published to the driver. Drivers display logs generated from its tasks/actors to its stdout and stderr.
+
+Let's look at a code example to see how this works.
+
+.. code-block:: python
+
+    import ray
+    # Initiate a driver.
+    ray.init()
+
+    @ray.remote
+    def task():
+        print("task")
+    
+    ray.get(task.remote())
+        
+You should be able to see the string `task` from your driver stdout. 
+
+When logs are printed, the process id (pid) and an IP address of the node that executes tasks/actors are printed together. Check out the output below.
+
+.. code-block:: bash
+
+    (pid=45601) task
+
+How to set up loggers
+~~~~~~~~~~~~~~~~~~~~~
+When using ray, all of the tasks and actors are executed remotely in Ray's worker processes. 
+Since Python logger module creates a singleton logger per process, loggers should be configured on per task/actor basis. 
+
+.. note::
+
+    Note that to stream logs to a driver, they should be flushed to stdout and stderr.
+
+.. code-block:: python
+
+    import ray
+    import logging
+    # Initiate a driver.
+    ray.init()
+
+    @ray.remote
+    class Actor:
+        def __init__(self):
+            # Basic config automatically configures logs to
+            # be streamed to stdout and stderr.
+            logging.basicConfig(level=logging.DEBUG)
+        
+        def log(self, msg):
```

Comment:
This doesn't address how to setup up logging for tasks (I'm not actually sure how to do this)

Reply:
Done.